components/color-picker/demo/format.tsx

```diff
@@ -0,0 +1,57 @@
+import { Col, ColorPicker, Row, Space } from 'antd';
+import type { Color } from 'antd/lib/color-picker';
+import React, { useMemo, useState } from 'react';
+
+export default () => {
+  const [colorHex, setColorHex] = useState<Color | string>('#1677ff');
+  const [colorHsb, setColorHsb] = useState<Color | string>('hsb(215,91%,100%)');
+  const [colorRgb, setColorRgb] = useState<Color | string>('rgb(22,119,255)');
+
+  const hexString = useMemo(
+    () => (typeof colorHex === 'string' ? colorHex : (colorHex as Color).toHexString()),
+    [colorHex],
+  );
+  const hsbString = useMemo(
+    () => (typeof colorHsb === 'string' ? colorHsb : (colorHsb as Color).toHsbString()),
+    [colorHsb],
+  );
+  const rgbString = useMemo(
+    () => (typeof colorRgb === 'string' ? colorRgb : (colorRgb as Color).toRgbString()),
+    [colorRgb],
+  );
```

Comment:
```suggestion
  const hexString = useMemo(
    () => (typeof colorHex === 'string' ? colorHex : colorHex.toHexString()),
    [colorHex],
  );
  const hsbString = useMemo(
    () => (typeof colorHsb === 'string' ? colorHsb : colorHsb.toHsbString()),
    [colorHsb],
  );
  const rgbString = useMemo(
    () => (typeof colorRgb === 'string' ? colorRgb : colorRgb.toRgbString()),
    [colorRgb],
  );
```
这里应该不用类型断言了，前面有 `typeof` 会自动窄化类型的

Reply:
确实，写多了，我优化一下